Source/API/galapagos_metadata.h

```diff
@@ -9,10 +9,11 @@
 
 typedef std::function<double(creature* creature)> fitness_func_t;
 
-struct crossover_metadata { };
-struct mutation_metadata { };
-struct selection_algorithm_metadata { };
-struct termination_condition_metadata { };
+// It seems base struct need at least 1 virtual member to be considered polymorphic
```

Comment:
Why are you putting a `virtual` destructor in the metadata structs?

Reply:
It is important for the metadata to be polymorphic so the core assembly (Galapagos.dll) can pass around the metadata as the polymorphic base class and it will never have to know about concrete implementations. Unfortunately c++ objects (structs or classes) are not considered polymorphic unless the have at least 1 virtual member. To get around this I added a virtual destructor with default implementation to each metadata struct.